contracts/MechMarketplace.sol

```diff
@@ -365,6 +371,29 @@ contract MechMarketplace is IErrorsMarketplace {
         emit SetMechFactoryStatuses(mechFactories, statuses);
     }
 
+    // Check and escrow delivery rate
+    function _checkAndEscrowDeliveryRate(address mech) internal {
+        IMech.MechType mechType = IMech(mech).mechType();
+        uint256 maxDeliveryRate = IMech(mech).maxDeliveryRate();
+
+        // Check delivery rates
+        if (mechType == IMech.MechType.FixedPrice) {
+            // Check the request delivery rate for a fixed price
+            if (msg.value < maxDeliveryRate) {
+                revert InsufficientBalance(msg.value, maxDeliveryRate);
+            }
+        } else {
+            // Check that there is no incoming deposit
+            if (msg.value > 0) {
+                revert NoDepositAllowed(msg.value);
+            }
+
+            // TODO Add onERC1155Received to the receiving contract
+            // Get max subscription rate for escrow from sender
+            IERC1155(subscriptionNFT).safeTransferFrom(msg.sender, address(this), subscriptionTokenId, maxDeliveryRate, "");
```

Comment:
Where is the subscriptionNFT/subscriptionTokenId defined ?

Reply:
Outside of the scope